h2/utilities.py

```diff
@@ -197,10 +197,10 @@ def validate_headers(headers, hdr_validation_flags):
     headers = _reject_surrounding_whitespace(
         headers, hdr_validation_flags
     )
-    headers = _reject_te(
+    headers = _strip_connection_header(
```

Comment:
I wonder if we should break this out into a separate `normalize_headers` pipeline, enabled by default, but which can then be disabled separately if necessary.

RFC § 8.1.2.2 also says:

> HTTP/2 does not use the Connection header field to indicate
   connection-specific header fields; in this protocol, connection-
   specific metadata is conveyed by other means.  An endpoint MUST NOT
   generate an HTTP/2 message containing connection-specific header
   fields; any message containing connection-specific header fields MUST
   be treated as malformed (Section 8.1.2.6).

and you only encounter them and remove them if you’re interworking from HTTP/1.x to HTTP/2. If I’m communicating with an endpoint that speaks pure HTTP/2, I might want to raise a ProtocolError if I get these bad headers.

Or maybe we should add a different configuration option "I expect to get messages that came from HTTP/1.x"? Less sure about that.